### PR TITLE
fix: handle missing route options

### LINKED ISSUE #232: Cannot read properties of undefined (reading 'pressureHandler')
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.13.0

### Plugin version

8.5.1

### Node.js version

v18.20.3

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

14.5 Sonoma

### Description

I upgraded under-pressure plugin from 8.1 to the latest (8.5.1) and it failed with error 
```
{
    "type": "TypeError",
     "message": "Cannot read properties of undefined (reading 'pressureHandler')",
     "stack":
          TypeError: Cannot read properties of undefined (reading 'pressureHandler')
              at Object.onRequest (/Users/tweekus/projects/platformeco/platformeco-fastify-connector/node_modules/@fastify/under-pressure/index.js:224:54)
              at hookIterator (/Users/tweekus/projects/platformeco/platformeco-fastify-connector/node_modules/fastify/lib/hooks.js:251:10)
              at next (/Users/tweekus/projects/platformeco/platformeco-fastify-connector/node_modules/fastify/lib/hooks.js:179:16)
              at handleResolve (/Users/tweekus/projects/platformeco/platformeco-fastify-connector/node_modules/fastify/lib/hooks.js:190:5)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
}
```

### Link to code that reproduces the bug

I run tests for different versions of fastify server and it failed for 4.0.0-4.22.0 with error

### Expected Behavior

I expects work like in 8.1 version

### PR DESCRIPTION
**Pull request by c0sx**
Fixes #232 

in fastify < 4.10  req in onRequest hook didn't have `routeOptions` at all - [PR](https://github.com/fastify/fastify/pull/4397)
in fastify < 4.23 `routeOptions` didn't have `config` property - [PR](https://github.com/fastify/fastify/pull/4470)

I handled both cases to keep compatibility, but I don't know how add tests for it. Do you have any suggestions?

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -220,8 +220,17 @@ async function fastifyUnderPressure (fastify, opts) {
     return false
   }
 
+  function getReqConfig(req) {
+    if (req.routeOptions && req.routeOptions.config) {
```
Comment (jean-michelet) [inline index.js:224]:
```suggestion
    if (req.routeOptions?.config) {
```

Comment (jean-michelet) [reply index.js:224]: I don't think we need a function for this.

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
     "@types/node": "^20.1.0",
-    "fastify": "^4.0.0-rc.2",
+    "fastify": "^4.0.0",
```
Comment (jean-michelet) [inline package.json:37]: Why?

**Review by jean-michelet — COMMENTED** [review]
I am unable to reproduce the bug from the repository you linked in the issue.
You should first write a test that fails in order to prove the existence of the bug, then write the fix that make it pass.

File: index.js (hunk 1 above)
Comment (c0sx) [reply index.js:224]: `?.` is allowed? package.json doesn't have `engine` and code uses `||` for defaults

**Comment by c0sx** [comment]
> I am unable to reproduce the bug from the repository you linked in the issue. You should first write a test that fails in order to prove the existence of the bug, then write the fix that make it pass.

Please check fastify version. It should be 4.0.0-4.22.0 to reproduce the bug. And I can't write tests because of it.

File: package.json (hunk 2 above)
Comment (c0sx) [reply package.json:37]: Sorry, I forget to return it back

File: index.js (hunk 1 above)
Comment (jean-michelet) [reply index.js:224]: Well, I started to contribute to Fastify for v5 so I am not very sure.
But for v4, [I think it's ok too](https://fastify.dev/docs/v4.28.x/Reference/LTS/#schedule).

**Comment by jean-michelet** [comment]
When creating a reproduction repository, it's important to specify the **exact version** where the bug occurs (e.g., `4.0.0`), rather than using version ranges (`^4.0.0`) in package.json. If several versions are involved, you can write a short README to explain the steps to reproduce the bugs.

File: index.js [hunk 3]
```diff
@@ -220,8 +220,17 @@ async function fastifyUnderPressure (fastify, opts) {
     return false
   }
 
+  function getReqConfig(req) {
```
Comment (jean-michelet) [inline index.js:223]: Still not sure this function is necessary, but I'm not radically opposed to it.

File: index.js (hunk 1 above)
Comment (gurgunday) [reply index.js:224]: > ?. is allowed?

Yes

Comment (gurgunday) [reply index.js:224]: Oh this is a backport, then no

**Review by jean-michelet — APPROVED** [review]
LGTM

**Comment by jean-michelet** [comment]
I think it is possible to run versioned tests in a child process, but I am not sure that this is something we usually do.

**Comment by c0sx** [comment]
> I think it is possible to run versioned tests in a child process, but I am not sure that this is something we usually do.

It could be useful, but I'm not sure too.

Usually I'm doing this with docker-compose, but I it requires changes in CI pipeline.

**Comment by gurgunday** [comment]
Yeah https://github.com/fastify/under-pressure/pull/228 was breaking for v8 (Fastify v4 line) actually, it should have considered the then-deprecated setting as well instead of just changing to routeOptions

**Review by gurgunday — APPROVED** [review]

**Comment by jean-michelet** [comment]
Can we merge this?

**Review by gurgunday — APPROVED** [review]

**Pull request merged**